core/src/main/java/hudson/Util.java

```diff
@@ -1688,6 +1689,35 @@ public static void closeAndLogFailures(@CheckForNull Closeable toClose, @Nonnull
         }
     }
 
+    @Restricted(NoExternalUse.class)
+    public static int permissionsToMode(Set<PosixFilePermission> permissions) {
+        PosixFilePermission[] allPermissions = PosixFilePermission.values();
+        int result = 0;
+        for (int i = 0; i < allPermissions.length; i++) {
+            result <<= 1;
+            result |= permissions.contains(allPermissions[i]) ? 1 : 0;
+        }
```

Comment:
If the bitwise operations here are too opaque I am happy to introduce constants for clarity.

Reply:
You wrote `testModeToPermissions`, so good enough for me.